File: ruby/leetcode/toeplitz-matrix.rb
# @param {Integer[][]} matrix
# @return {Boolean}
def is_toeplitz_matrix(matrix)
    for i in 0..(matrix.length - 1) do
        max_check = if i == 0 then matrix[0].length - 1 else 1 end
        for j in 0..max_check do
            val = matrix[i][j]
            c = j + 1
            r = i + 1
            while (r < matrix.length && c < matrix[0].length)
                if matrix[r][c] != val
                    return false
                end
                c += 1
                r += 1
            end
        end
    end
    return true
end

File: python/leetcode/toeplitz-matrix.py
class Solution:
    def isToeplitzMatrix(self, matrix):
        """
        :type matrix: List[List[int]]
        :rtype: bool
        """
        for i in range(len(matrix) - 1):
            max_check = len(matrix[0]) - 1 if i == 0 else 1
            for j in range(max_check):
                val = matrix[i][j]
                c = j + 1
                r = i + 1
                while r < len(matrix) and c < len(matrix[0]):
                    if matrix[r][c] != val:
                        return False
                    c += 1
                    r += 1
        return True
